Spryker/Sniffs/Commenting/DisallowArrayTypeHintSyntaxSniff.php

```diff
@@ -375,4 +387,112 @@ protected function containsArrayTypeNode(array $types): bool
 
         return false;
     }
+
+    /**
+     * @param \PHPStan\PhpDocParser\Ast\Type\TypeNode $typeNode
+     *
+     * @return bool
+     */
+    protected function isArrayTypeNode(TypeNode $typeNode): bool
+    {
+        return $typeNode instanceof ArrayTypeNode &&
+            ($typeNode->type instanceof IdentifierTypeNode || $typeNode->type instanceof ArrayTypeNode);
+    }
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile
+     * @param \SlevomatCodingStandard\Helpers\Annotation\VariableAnnotation|\SlevomatCodingStandard\Helpers\Annotation\ParameterAnnotation|\SlevomatCodingStandard\Helpers\Annotation\ReturnAnnotation|\SlevomatCodingStandard\Helpers\Annotation\ThrowsAnnotation|\SlevomatCodingStandard\Helpers\Annotation\PropertyAnnotation|\SlevomatCodingStandard\Helpers\Annotation\MethodAnnotation|\SlevomatCodingStandard\Helpers\Annotation\TemplateAnnotation|\SlevomatCodingStandard\Helpers\Annotation\ExtendsAnnotation|\SlevomatCodingStandard\Helpers\Annotation\ImplementsAnnotation|\SlevomatCodingStandard\Helpers\Annotation\UseAnnotation|\SlevomatCodingStandard\Helpers\Annotation\MixinAnnotation|\SlevomatCodingStandard\Helpers\Annotation\TypeAliasAnnotation|\SlevomatCodingStandard\Helpers\Annotation\TypeImportAnnotation $annotation
+     *
+     * @return void
+     */
+    protected function fixGenericObjectCollection(File $phpcsFile, Annotation $annotation): void
+    {
+        $typeNode = AnnotationHelper::getAnnotationTypes($annotation)[0];
+
+        $genericType = null;
+        $arrayType = null;
+        $unionTypes = [];
+
+        if (!$typeNode instanceof UnionTypeNode) {
+            return;
+        }
+
+        foreach ($typeNode->types as $type) {
+            if ($this->isGenericObject($type)) {
+                if ($genericType !== null) {
+                    return;
+                }
+
+                $genericType = (string)$type;
+
+                continue;
+            }
+
+            if ($this->isArrayTypeNode($type) && $type instanceof ArrayTypeNode) {
+                if ($arrayType !== null) {
+                    return;
+                }
+                $arrayType = $this->convertTypeToString($type->type);
+
+                continue;
+            }
+
+            $unionTypes[] = (string)$type;
+        }
+
+        if (!$genericType || !$arrayType) {
+            return;
+        }
+
+        $fix = $phpcsFile->addFixableError(
+            sprintf(
+                'Usage of old type hint syntax for generic in "%s" is disallowed, use generic type hint syntax instead.',
```

Comment:
```suggestion
                'Usage of old type hint syntax for generic in `%s` is disallowed, use generic type hint syntax instead.',
```
Please use code fencing instead of quoting for code elements.